### PR TITLE
Add parser for /proc/self/mountinfo and new combiner mounts

### PR DESCRIPTION
**Pull request by JoySnow**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:
This `/proc/self/mountinfo` can provide additional info that `/bin/mount` and `/proc/mounts` don't collect.

### CODE REVIEW

**Comment by JoySnow** [comment]
Hi @xiangce , could you help review this MR? We can discuss on the details of this adding part if any other thoughts.

File: insights/parsers/mount.py [hunk 1]
```diff
@@ -304,6 +328,75 @@ def _parse_mounts(self, content):
...
+        Please refer to its super-class :class:`MountedFileSystems` for more details.
+
+    The typical content of ``/proc/self/mountinfo`` file looks like::
+
+        39 0 253:0 / / rw,relatime shared:1 - xfs /dev/mapper/rootvg-lvlocroot rw,attr2,inode64,noquota
+        47 39 8:1 / /boot rw,relatime shared:30 - xfs /dev/sda1 rw,attr2,inode64,noquota
+        65 39 253:19 / /data rw,relatime shared:44 - ext4 /dev/mapper/vgdata-lvdata rw,data=ordered
+        58 39 253:15 / /opt rw,relatime shared:45 - xfs /dev/mapper/rootvg-lvlocopt rw,attr2,inode64,noquota
+        61 39 253:17 / /home rw,nosuid,relatime shared:46 - xfs /dev/mapper/rootvg-lvlochome rw,attr2,inode64,noquotao
+
```
Comment (psachin) [inline insights/parsers/mount.py:346]: It will be helpful to the rule developer if you can document the returned attributes.

**Review by psachin — APPROVED** [review]

**Comment by JoySnow** [comment]
Hi @xiangce , could you help review this MR?          

BTW, according to the CI doc-test , it reports this [error](https://github.com/RedHatInsights/insights-core/runs/6320487495?check_suite_focus=true#step:5:163 ) :        
```
/home/runner/work/insights-core/insights-core/insights/combiners/mounts.py:docstring of insights.combiners.mounts.MountAddtlInfo.fs_freq:1:duplicate object description of insights.combiners.mounts.MountAddtlInfo.fs_freq, other instance in shared_combiners_catalog/mounts, use :noindex: for one of them
``` 
Actually, I only mentioned  attribute `fs_freq` once in this combiner.        
Do you know how to fix this error here?

**Comment by JoySnow** [comment]
Updated related parsers and combiner according to discussion with @xiangce .       
Could you help recheck this MR ?

**Review by xiangce — APPROVED** [review]

**Pull request merged**